packages/block-library/src/table/edit.js

```diff
@@ -400,18 +458,53 @@ export default class TableEdit extends Component {
 				</BlockControls>
 				<InspectorControls>
 					<PanelBody title={ __( 'Table Settings' ) } className="blocks-table-settings">
+						<div className="block-library-table__dimensions__row">
+							<TextControl
+								type="number"
+								className="block-library-table__dimensions__width"
+								label={ widthUnit ? `${ __( 'Width' ) } (${ widthUnit })` : __( 'Width' ) }
```

Comment:
I would move this ternary out of the component itself and put it in a variable above, to improve readability. Because this line is quite intended, it'll be easier to read above.

As @afercia mentioned, this won't work for l10n because it assumes English order/characters for this string. Instead, you'll want:

```js
import { __, sprintf } from '@wordpress/i18n';

// […]
const widthLabel = widthUnit ? sprintf( __( 'Width (in %s)' ), widthUnit ) : __( 'Width' );

<TextControl
  label={ widthLabel }
/>
```

Reply:
Thanks both of you for catching that - not sure why my brain blipped on using `sprintf`. Pushed a commit to fix it.